contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,853 @@
+a Ready Condition, if `.Spec.Replicas` is equal to `.Status.Replicas`, and if 
+there are Pods that do not match the 
+[declared desired PodTemplate](#podtemplate-selection), the Pod corresponding to 
+the largest ordinal will be deleted.
+1. If the StatefulSet controller has achieved the 
+[declared target state](#target-state), and if that state has a 
+`.Spec.ParitionOrdinal` equal to `0`, the StatefulSet controller will 
+[complete any in progress updates](#update-completion).
+1. The controller will [report its status](#status-reporting).
+1. The controller will perform any necessary
+[maintenance of its revision history](#history-maintenance).
+
+### StatefulSet Revision History
+The StatefulSet controller will use labeled, versioned PodTemplates to keep a 
+history of updates preformed on a StatefulSet. The number of stored PodTemplates 
+is considered to be the limit of the StatefulSet's revision history. The 
+maximum revision history limit for a StatefulSet must be at least two, but it 
+may be greater.
+
+#### PodTemplate Creation
+When the StatefulSet controller creates a PodTemplate for a StatefulSet, it will 
+do the following.
+
+1. The controller will set the PodTemplate's `.PodTemplateSpec` field to the 
+StatefulSet's `.Spec.Template` field.
+1. The controller will create a ControllerRef object in the PodTemplate's 
+`.OwnerReferences` list to mediate selector overlapping.
+1. The controller will label the PodTemplate with a 
+`StatefulSetPodTemplateLabel` set to the StatefulSet's `.Name` to allow for 
+selection of the PodTemplates that comprise the StatefulSet's revision history.
+1. The controller will label the PodTemplate with a 
+`StaefulSetTemplateGenerationLabel` set to the StatefulSet's 
+`.Spec.TemplateGeneration`.
+1. The controller will set the Name of the PodTemplate to a concatenation of the 
+`.Name` of the StatefulSet and the `.Spec.TemplateGeneration`.
+1. The controller will then create the PodTemplate.
+
+#### PodTemplate Deletion
+When the `StatefulSet` controller deletes a PodTemplate in the revision 
+history of a StatefulSet it will do the following.
+
+1. If the PodTemplate's ControllerRef does not match the StatefulSet, the 
+controller will not delete the PodTemplate. In this way, we prevent selector 
+overlap from causing the deletion of PodTemplates that are part of another 
+object's revision history. In practice, these PodTemplates will be filtered out
+prior to history maintenance.
+1. If the PodTemplate's ControllerRef matches the StatefulSet, the 
+StatefulSet controller orphan the PodTemplate by removing its ControllerRef, 
+and it will allow the PodTemplate to be deleted via garbage collection.
+
+#### History Reconstruction
+In order to reconstruct the history of revisions to a StatefulSet, the 
+StatefulSet controller will do the following.
+
+1. If the StatefulSet's `.Spec.TemplateGeneration` is nil, the StatefulSet 
```

Comment:
Users should be able to initialize this field but not update it. Not sure if you mention this somewhere in this doc.